src/api_generation/high_level.jl

```diff
@@ -1,3 +1,21 @@
+abstract type OperationType end
+struct RestBased <: OperationType end
+struct JSONBased <: OperationType end
+
+function _parse_protocol(function_name::AbstractString, protocol::AbstractString)
+    if protocol in ("json", "query", "ec2")
+        return JSONBased()
+    elseif protocol in ("rest-xml", "rest-json")
+        return RestBased()
+    else
+        throw(
+            ProtocolNotDefined(
+                "$function_name is using a new protocol; $protocol which is not supported."
```

Comment:
I am not sure the word "new" should be here.
since someone might for example be using this code many years after the protocol was added.
(say they were pinned to an old version for some reason)

idk maybe
```suggestion
                "$function_name is using an unexpected protocol; $protocol which is not supported."
```

Reply:
This is used for generating high-level APIs. End users shouldn't care for this, as it only matters for maintainers of this package.